# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/linux/launch/SandboxLaunch.cpp
# Commit: 10e01cf10faf
# Full Hash: 10e01cf10fafca4a150c2f76b3ce4e33bbd06223
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2020-03-09 21:52:54
# Regressor Bug: 1608558
# File Overlap Count: 1
# Description:
#   Bug 1608558 - pt1 - add linux sandboxing to socket process. r=gcp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60014
# ==============================================================================

diff -r 61e14505a488 -r 10e01cf10faf security/sandbox/linux/launch/SandboxLaunch.cpp
--- a/security/sandbox/linux/launch/SandboxLaunch.cpp	Fri Mar 06 20:29:47 2020 +0000
+++ b/security/sandbox/linux/launch/SandboxLaunch.cpp	Mon Mar 09 14:56:03 2020 +0000
@@ -249,6 +249,8 @@
       return 0;
     case GeckoProcessType_RDD:
       return PR_GetEnv("MOZ_DISABLE_RDD_SANDBOX") == nullptr ? 1 : 0;
+    case GeckoProcessType_Socket:
+      return PR_GetEnv("MOZ_DISABLE_SOCKET_PROCESS_SANDBOX") == nullptr ? 1 : 0;
     default:
       return 0;
   }
@@ -296,6 +298,12 @@
   }
 
   switch (aType) {
+    case GeckoProcessType_Socket:
+      if (level >= 1) {
+        canChroot = true;
+        flags |= CLONE_NEWIPC;
+      }
+      break;
     case GeckoProcessType_GMPlugin:
     case GeckoProcessType_RDD:
       if (level >= 1) {